### PR TITLE
[MINOR][Doc][SQL][Core] Fix typo in document and comments

### PR DESCRIPTION
**Pull request by kiszk**
### What changes were proposed in this pull request?

Fixed typo in `docs` directory and in other directories

1. Find typo in `docs` and apply fixes to files in all directories
2. Fix `the the` -> `the`

### Why are the changes needed?

Better readability of documents

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

No test needed

### CODE REVIEW

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -220,7 +220,7 @@ license: |
...
+  - Since Spark 3.0, when casting string value to integral types(tinyint, smallint, int and bigint), datetime types(date, timestamp and interval) and boolean type, the leading and trailing whitespaces(<= ASCII 32) will be trimmed before converted to these type values, e.g. `cast(' 1\t' as int)` results `1`, `cast(' 1\t' as boolean)` results `true`, `cast('2019-10-10\t as date)` results the date value `2019-10-10`. In Spark version 2.4 and earlier, while casting string to integrals and booleans, it will not trim the whitespaces from both ends, the foregoing results will be `null`, while to datetimes, only the trailing spaces(= ASCII 32) will be removed.
```
Comment (srowen) [inline docs/sql-migration-guide.md]: Nit: add spaces before parens

File: docs/sql-ref-null-semantics.md [hunk 2]
```diff
@@ -236,7 +236,7 @@ The comparison operators and logical operators are treated as expressions in
 Spark. Other than these two kinds of expressions, Spark supports other form of
 expressions such as function expressions, cast expressions, etc. The expressions
 in Spark can be broadly classified as :
-- Null in-tolerent expressions
+- Null in-tolerant expressions
```
Comment (srowen) [inline docs/sql-ref-null-semantics.md]: intolerant

**Review by srowen — COMMENTED** [review]
Looks good

File: docs/sql-ref-null-semantics.md (hunk 2 above)
Comment (kiszk) [reply docs/sql-ref-null-semantics.md]: Fixed. Also fixed other usages in the same file.

File: docs/sql-migration-guide.md (hunk 1 above)
Comment (kiszk) [reply docs/sql-migration-guide.md]: thanks, done

**Comment by dongjoon-hyun** [comment]
Jenkins seems to be out of order.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by srowen** [comment]
@kiszk one more random one: in DecisionTreeMetadata.scala, there's a message with "Considering remove" in it. Change to "Consider removing"?

**Comment by dongjoon-hyun** [comment]
Since this is a doc and comment-only change, I'll merge this PR. 
Although Jenkins isn't triggered, `GitHub Action` building and linters are passed.
Thank you, @kiszk and @srowen .

**Pull request closed without merging**